### PR TITLE
[supply][action] add `download_apk_from_google_play` action (and corresponding `supply` methods)

### PR DESCRIPTION
**Pull request by AliSoftware**
## Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] ~I've updated the documentation if necessary.~ N/A; actions documentation are auto-generated anyway

## Motivation and Context

This adds a new `download_apk_from_google_play` action to download the Universal Code-Signed APK generated by Google after you uploaded the AAB.

This is especially useful when you use [Play App Signing](https://support.google.com/googleplay/android-developer/answer/9842756) to sign your app, because in those cases you'll typically use the `upload_to_google_play` action to upload a `.aab` to Google Play Console (GPC), and let Google generate and code-sign the `.apk` files for that `.aab` (including split APKs for various architectures, but more importantly the Universal one compatible with all devices)

In those situations of using "Play App Signing", you can't use `bundletool` to generate the signed `.apk` from the `.aab` yourself because you don't have the code signing key (only Google has it), so the only option is to download the `.apk` from GPC.
And it's useful to get the universal, codesigned APK back because we can use it for running UI tests on it, making sure we run those on the _exact_ APK that will be ultimately shipped to end users (including testing things that rely on code-signing signature checks for security and integrity validation).

## Description

 - `Supply::Client`
   - Added the `list_generated_universal_apks(package_name:, version_code:)` method on `Supply::Client` to be able to call [the `generatedapks.list` API endpoint](https://developers.google.com/android-publisher/api-ref/rest/v3/generatedapks/list)
   - Added the `download_generated_universal_apk(generated_universal_apk:, destination:)` method on `Supply::Client` to be able to call [the `generatedapks.download` API endpoint](https://developers.google.com/android-publisher/api-ref/rest/v3/generatedapks/download)
   - Added those entry points to the `spec/client_spec.rb` tests
   - Added a `GeneratedUniversalApk` model to allow manipulating the result of those methods more easily
- New Action
   - Added the `download_universal_apk_from_google_play` action as well as its unit tests (spec)

## Testing Steps

Run the new Unit Tests:
```
bundle exec rspec fastlane/spec/actions_specs/download_universal_apk_from_google_play_spec.rb supply/spec/client_spec.rb
```

### CODE REVIEW

**Comment by AliSoftware** [comment]
@joshdholtz Now that you're back from your vacation 😃  May I kindly ask a review of this one? I'd love to be able to finally use it in our automation soon 🙂

File: supply/spec/client_spec.rb [hunk 1]
```diff
@@ -40,6 +40,8 @@
         expect(subject.class.method_defined?(:get_edit_listing)).to eq(true)
         expect(subject.class.method_defined?(:list_edit_apks)).to eq(true)
         expect(subject.class.method_defined?(:list_edit_bundles)).to eq(true)
+        expect(subject.class.method_defined?(:list_generatedapks)).to eq(true)
+        expect(subject.class.method_defined?(:download_generatedapk)).to eq(true)
```
Comment (getaaron) [inline supply/spec/client_spec.rb:44]: Are these correct?

Comment (AliSoftware) [reply supply/spec/client_spec.rb:44]: Yes. That spec tests methods we expect to be defined **on the `AndroidPublisher::AndroidPublisherService` internal client** (not to be confused with the `Supply::Client`).

Those two particular ones that I added there are documented here in the `AndroidPublisherService` ruby doc:
 - [`AndroidPublisher::AndroidPublisherService#list_generatedapks`](https://googleapis.dev/ruby/google-apis-androidpublisher_v3/v0.33.0/Google/Apis/AndroidpublisherV3/AndroidPublisherService.html#list_generatedapks-instance_method)
 - [`AndroidPublisher::AndroidPublisherService#download_generatedapk`](https://googleapis.dev/ruby/google-apis-androidpublisher_v3/v0.33.0/Google/Apis/AndroidpublisherV3/AndroidPublisherService.html#download_generatedapk-instance_method)

And those are the ones that I call in the `Supply::Client` corresponding wrapper methods [here](https://github.com/fastlane/fastlane/pull/21315/files#diff-f291ca0a82f6a4ac1725fec2caa16f022ec8a2852841fa13f3016dea73584484R316) and [here](https://github.com/fastlane/fastlane/pull/21315/files#diff-f291ca0a82f6a4ac1725fec2caa16f022ec8a2852841fa13f3016dea73584484R332), respectively.

So all that is correct (and consistent with how the rest of the spec and that file is written) 🙂

Comment (getaaron) [reply supply/spec/client_spec.rb:44]: Ah I see. Thanks! I'm much less familiar with the Android code.

**Review by getaaron — APPROVED** [review]

File: supply/spec/client_spec.rb (hunk 1 above)
Comment (AliSoftware) [reply supply/spec/client_spec.rb:44]: No worries! Took me some time to familiarize myself with the AndroidPublisher API and Supply code and specs as well when I wrote that PR 😅 🙂

**Pull request merged**